x/schema/keeper/schema.go

```diff
@@ -1,14 +1,62 @@
 package keeper
 
 import (
+	"bytes"
 	"encoding/binary"
+	"encoding/json"
+	"os"
+	"path/filepath"
 
 	"github.com/cosmos/cosmos-sdk/store/prefix"
 	sdk "github.com/cosmos/cosmos-sdk/types"
 	"github.com/google/uuid"
+	shell "github.com/ipfs/go-ipfs-api"
 	"github.com/sonr-io/sonr/x/schema/types"
 )
 
+const (
+	URL_PERSISTENCE_READ = "ipfs.Sonr.ws"
+)
+
+var (
+	ipfs_inter = shell.NewShell(URL_PERSISTENCE_READ)
+)
+
+func (k Keeper) LookUpContent(cid string, content interface{}) error {
+	out_path := filepath.Join(os.TempDir(), cid+".txt")
+	err := ipfs_inter.Get(cid, out_path)
+
+	if err != nil {
+		return err
+	}
+
+	resp, err := os.ReadFile(out_path)
+
+	if err != nil {
+		return err
+	}
+
+	if err = json.Unmarshal(resp, &content); err != nil {
+		return err
+	}
+
+	if err != nil {
+		return err
+	}
+
+	defer os.Remove(out_path)
+
+	return nil
+}
+
+func (k Keeper) PinContent(payload interface{}) (string, error) {
+	b, err := json.Marshal(payload)
+	if err != nil {
+		return "", err
+	}
+	return ipfs_inter.Add(bytes.NewReader(b))
+}
+
```

Comment:
this probably won't work as the endpoint for uploading is different from downloading. One of these two functions will probaly break.

Reply:
i was under the impression we had seperate domains for writing and reading. ill change that

Reply:
We do, I thought ipfs.add uses the same endpoint for both

Reply:
updated to use two shells